library/core/src/lib.rs

```diff
@@ -412,13 +412,13 @@ pub mod primitive;
     dead_code,
     unused_imports,
     unsafe_op_in_unsafe_fn,
-    ambiguous_glob_reexports
+    ambiguous_glob_reexports,
+    deprecated_in_future
```

Comment:
This was only *with* `bootstrap` before, so I think it should go away.

Reply:
stdarch isn't upgraded (see CI errors above) in these PRs, so until that happens I'm pretty sure this needs to stay.

Reply:
OK, I don't have the full picture of what that's guarding. It just looked weird that a `cfg_attr(bootstrap, ..)` got "promoted" this way.

Reply:
I suspect what's happening here is that it used to only be necessary at cfg(bootstrap) but now is always needed, because of something in cfg(bootstrap) territory? It's not obvious to me how deprecated_in_future _could_ depend on bootstrap in the first place though.